content/post/2020-06-01-macos-installer/index.Rmd

```diff
@@ -0,0 +1,265 @@
+      echo = TRUE
+    )
+    cast <- asciicast::read_cast(tmp)
+  
+    dir.create(
+      dirname(filename), 
+      showWarnings = FALSE, 
+      recursive = TRUE
+    )
+
+    asciicast::write_gif(
+      cast, 
+      filename, 
+      rows = options$rows %||% "auto", 
+      theme = "asciinema", loop = 1,
+      speed = options$speed %||% 1.0
+    )
+  }
+  
+  extra <- knitr::knit_hooks$get('plot')(filename, options)
+  knitr::engine_output(options, options$code, '', extra)
+})
+```
+
+## TL;DR
+
+If you need to use multiple versions of R on macOS, look at the
+`install-rstats` package in the 'Installer' section below.
+
+## Introduction
+
+As an R package developer, it is mostly your choice which R versions
+you support. If your package is on CRAN, then you must support the
+latest R release and also fix issues that come up when your package is
+tested on the current development version (R-devel). These two
+versions are the minimum, but developers typically try to do better.
+E.g. the tidyverse supports the [four previous versions or
+R](https://www.tidyverse.org/blog/2019/04/r-version-support/).
+
+The more versions you support, the more versions you typically keep on
+your development machine, to be able to reproduce problems and fix
+bugs, that are specific to older or newer versions. [R-hub is
+great](https://blog.r-hub.io/2019/03/26/why-care/) for quick checks,
+but if there is a problem, you'll want to run R on your local machine
+to investigate. In this post we show a way to run multiple R versions
+on macOS, at the same time.
+
+## The macOS R installer
+
+The official CRAN R installers work differently on different operating
+systems. On Windows, every R version is installed into its own
+directory and works independently of the others. On Linux,
+distributions typically only support one R version at any time, but
+you can easily use Docker to run [multiple R
+versions](https://github.com/r-hub/r-minimal#readme).
+
+macOS is a bit more difficult. You can install [multiple (minor) R
+versions](https://cran.r-project.org/doc/manuals/R-admin.html#Multiple-versions),
+but by default you cannot use them independently.
+
+The macOS R installer creates a [macOS
+framework](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Frameworks.html)
+in `/Library/Frameworks/R.framework`. By default it removes the
+already installed version, so this directory only contains one version
+of the framework. To keep the previous version, you need to "forget"
+(`pkgutil --forget`) all Apple packages that the installer creates.
+See the [admin
+manual](https://cran.r-project.org/doc/manuals/R-admin.html#Multiple-versions).
+This way you can install multiple R versions, as long as they are
+different minor versions. I.e. you can install R 3.6.3 and R 4.0.0
+together, but not R 3.6.2 and R 3.6.3. After installing a couple of R
+versions, `/Library/Frameworks/R.framework` may look like this:
+
+```{bash}
```

Comment:
could you add alt text for this figure?

Alternatively maybe `fs::file_tree()` in an R chunk makes sense since it outputs Markdown?